soh/soh/Enhancements/randomizer/hint.cpp

```diff
@@ -3,9 +3,9 @@
 namespace Rando {
 Hint::Hint() : text(std::move(Text())) {}
 Hint::Hint(Text text_): text(std::move(text_)) {}
-Hint::Hint(Text text_, RandomizerCheck hintedLocation_, HintType hintType_, Text hintedRegion_)
+Hint::Hint(Text text_, RandomizerCheck hintedLocation_, HintType hintType_, std::string distributionName_, RandomizerArea hintedArea_)
     : text(std::move(text_)), hintedLocation(hintedLocation_), hintType(hintType_),
-      hintedRegion(std::move(hintedRegion_)) {
+      distribution(distributionName_), hintedArea(hintedArea_) { //not sure if removing std::move is correct, did it for safety, replace it if it's wise
```

Comment:
std::move would do nothing for an enum, so removing it was correct. To oversimplify a bit, When this was a Text object, it would mean that if you constructed a Text object when calling this constructor, it would prevent extra allocations and copies by just setting pointers. Pointers are the same size as ints/enum values so copying those around compared to moving a pointer are equally efficient.

Reply:
That being said you actually probably do want to `std::move` the `distributionName_`, since in a lot of cases here you're passing in a string literal, where std::move would make sense.